### PR TITLE
Feat/data mapper

### PR DESCRIPTION
**Pull request by nguyenvukhang**
strip all entities of their methods

### CODE REVIEW

File: packages/database/src/entity/DAG.ts [hunk 1]
```diff
@@ -5,11 +5,9 @@ import {
...
   JoinTable,
 } from 'typeorm'
-import { container } from '../data-source'
-
-import { User, Degree, Module } from '.'
-import { ModuleRepository } from '../repository/Module'
-import { DAGRepository } from '../repository/DAG'
+import { User } from './User'
+import { Degree } from './Degree'
+import { Module } from './Module'
```
Comment (nguyenvukhang) [inline packages/database/src/entity/DAG.ts:10]: Direct imports within `entity` folder for safety

File: packages/database/src/repository/DAG.ts [hunk 2]
```diff
@@ -91,7 +90,70 @@ async function initialize(props: DAGInitProps): Promise<void> {
   })
 }
 
+/**
+ * Toggle a Module's status between placed and hidden.
+ * @param {string} moduleCode
+ */
+async function toggleModule(thisDag: DAG, moduleCode: string): Promise<DAG> {
```
Comment (weiseng18) [inline packages/database/src/repository/DAG.ts:99]: this is my bad, we were deciding between mutating `this` and returning a new object, but here we have both, so don't need to `const res` and return I believe

File: packages/database/tests/dag.test.ts [hunk 3]
```diff
@@ -17,6 +19,17 @@ beforeAll(async () => {
...
+test('all imports are defined', () => {
+  const entities = [Module, Degree, User, DAG]
+  const repos = [UserRepository, DegreeRepository, DAGRepository]
+  entities.forEach((e) => {
+    expect(e).toBeDefined()
+  })
+  repos.forEach((e) => {
+    expect(e).toBeDefined()
+  })
+})
```
Comment (nguyenvukhang) [inline packages/database/tests/dag.test.ts:31]: check imports to ensure this doesn't happen again

**Review by weiseng18 — APPROVED** [review]
lgtm

**Pull request merged**